### DISCUSSION TITLE
Routes not working when using vercel.json

### DISCUSSION THREAD

**Discussion by reisenbauer** [vercel/vercel#6548] [Help] [answered]
Hello,

i'm having troubles getting the vercel.json file running:
- requesting **/api/db** is working and is passed to my serverless function
- requesting **/** returns 404 and does not serve the vue app.

```
# vercel.json
{
  "version": 2,
  "builds": [
    { "src": "api/index.js", "use": "@now/node" }
  ],
  "rewrites": [
    {
      "source": "/api/(.*)",
      "destination": "/api/index.js"
    }
  ]
}

```

```
# package.json
  "scripts": {
    "dev": "webpack-dev-server --inline --progress --config build/webpack.dev.conf.js",
    "start": "npm run dev",
    "build": "node build/build.js"
  },
```

anyone got a idea of how to fix this?

thanks,
Stefan

**Comment by mcsdevv** [comment]
Hey, @reisenbauer! Without the full code, it's always tricky, but from what you've shared you can just delete the `vercel.json` file entirely and rely on zero-configuration instead - this will add the required routes for you behind the scenes.

**Reply by reisenbauer** [reply]
Hi Matthew,
thanks for the reply - i've just created a new vercel project (template vue) and followed the guide here: https://vercel.com/docs/serverless-functions/introduction

after creating the api/index.js file, a call to http://127.0.0.1:3000/api/?name=example returned an error 404,
then i created the vercel.json as shown above and both (the vue app + the api/index.js) started working.

If you need any more details, let me know please!

thanks,
Stefan

**Reply by mcsdevv** [reply, marked as answer]
Hi Stefan, that's very strange, I was unable to reproduce and all tests continue to pass.

To remove any doubt, can you provide a link to the repository you're running this from?

**Reply by reisenbauer** [reply]
Hi Matthew, sorry for the delay - i've re-created the entire project and now it's working as expected!
thanks for your Support!
